src/plzinit/plugins.go

```diff
@@ -0,0 +1,285 @@
+// the host repo config file, and creating a build target in //plugins.
+func InitPlugins(plugins []string) {
+	log.Debug("Initialising plugin(s): %v", plugins)
+	for _, p := range plugins {
+		if err := initPlugin(p); err != nil {
+			log.Warningf("could not initialise plugin %s. Got error: %v", p, err)
+		}
+	}
+}
+
+func initPlugin(plugin string) error {
+	if err := createTarget("plugins/BUILD", plugin); err != nil {
+		return err
+	}
+	if err := injectPluginConfig(plugin); err != nil {
+		return err
+	}
+	return nil
+}
+
+func injectPluginConfig(plugin string) error {
+	configPath := path.Join(core.RepoRoot, ".plzconfig")
+	if _, err := os.Stat(configPath); errors.Is(err, os.ErrNotExist) {
+		log.Warningf("config file %s does not exist", configPath)
+		return err
+	}
+
+	configFile, err := os.Open(configPath)
+	if err != nil {
+		return err
+	}
+	defer configFile.Close()
+	file := ast.Read(configFile)
+	switch plugin {
+	case "python":
+		file = writePythonConfigFields(file)
+	case "java":
+		file = writeJavaConfigFields(file)
+	case "cc":
+		file = writeCCConfigFields(file)
+	default:
+		log.Fatalf("failed to initialise plugin. \"%s\" not recognised", plugin)
+	}
+	ast.Write(file, configPath)
+	return nil
+}
+
+func writePythonConfigFields(file ast.File) ast.File {
+	configMap := map[string]string{
+		"piptool":             "PipTool",
+		"pipflags":            "PipFlags",
+		"pextool":             "PexTool",
+		"defaultinterpreter":  "DefaultInterpreter",
+		"testrunner":          "TestRunner",
+		"debugger":            "Debugger",
+		"moduledir":           "ModuleDir",
+		"defaultpiprepo":      "DefaultPipRepo",
+		"wheelrepo":           "WheelRepo",
+		"wheelnamescheme":     "WheelNameScheme",
+		"interpreteroptions":  "InterpreterOptions",
+		"disablevendorflags":  "DisableVendorFlags",
+		"usepypi":             "UsePypi",
+		"testrunnerbootstrap": "TestrunnerDeps",
+	}
+
+	section := "Plugin"
+	subsection := "python"
+
+	// Check for existing python fields first
+	for _, s := range file.Sections {
+		if s.Key == section {
+			for _, field := range s.Fields {
+				if plugVal, ok := configMap[strings.ToLower(field.Name)]; ok {
+					file = ast.InjectField(file, plugVal, field.Value, section, subsection, false)
+				}
+			}
+		}
+	}
+
+	return file
+}
+
+func writeCCConfigFields(file ast.File) ast.File {
+	configMap := map[string]string{
+		"cctool":             "CCTool",
+		"cpptool":            "CPPTool",
+		"ldtool":             "LDTool",
+		"artool":             "ARTool",
+		"defaultoptcflags":   "DefaultOptCFlags",
+		"defaultdbgcflags":   "DefaultDbgCFlags",
+		"defaultoptcppflags": "DefaultOptCppFlags",
+		"defaultdbgcppflags": "DefaultDbgCppFlags",
+		"defaultldflags":     "DefaultLdFlags",
+		"pkgconfigpath":      "PkgConfigPath",
+		"testmain":           "TestMain",
+		"dsymtool":           "DsymTool",
+	}
+
+	return writeFieldsToConfig("cc", file, configMap)
```

Comment:
It seems that the cc plugin initialisation and migration is doing more than java and python, is this as expected?

Reply:
No not expected, sorry, just forgot to tidy up the java and python functions!